core/clients/6poc/web/views/mainView.jsx

```diff
@@ -0,0 +1,193 @@
+"use strict";
+
+var React = require('react');
+var $ = require('jquery');
+
+var MapView =  require('./mapView.jsx');
+
+module.exports = React.createClass({
+  getInitialState: function() {
+    var availableCategories = [
+      'aluminium',
+      'batteries',
+      'beverage_carton',
+      'bicycles',
+      'books',
+      'cans',
+      'cardboard',
+      'cds',
+      'chipboard',
+      'christmas_trees',
+      'clothes',
+      'computers',
+      'cooking_oil',
+      'cork',
+      'engine_oil',
+      'foil',
+      'furniture',
+      'glass',
+      'glass_bottles',
+      'green_waste',
+      'garden_waste',
+      'hardcode',
+      'hazardous_waste',
+      'light_bulbs',
+      'magazines',
+      'mobile_phones',
+      'music',
+      'newspaper',
+      'organic',
+      'paint',
+      'paper',
+      'paper_packaging',
+      'plastic',
+      'plastic_bags',
+      'plastic_bottles',
+      'plastic_packaging',
+      'polyester',
+      'printer_cartridges',
+      'rubble',
+      'scrap_metal',
+      'sheet_metal',
+      'shoes',
+      'tyres',
+      'waste',
+      'waste_oil',
+      'white_goods',
+      'wood']
+    return {result: null, mode: 0, condition: 0, radius: 50, availableCategories: availableCategories, categories: [], searchCpt: 0};
+  },
+  componentDidMount: function() {
+    var self = this;
+
+    var where = React.findDOMNode(this.refs.where);
```

Comment:
Not very react-y, but probably necessary for the interaction with google map autocomplete thing?